walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,111 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<TimeRange> availableTimeRanges = new ArrayList<TimeRange> ();
+
+    Collection<String> requestAttendees = request.getAttendees();
+    Collection<String> requestOptionalAttendees = request.getOptionalAttendees();
+    Set<String> requestAttendeesSet = new HashSet<> ();
+    Set<String> requestOptionalAttendeesSet = new HashSet<> ();
+    requestOptionalAttendeesSet.addAll(requestOptionalAttendees);
+    requestAttendeesSet.addAll(requestAttendees);
+    requestAttendeesSet.addAll(requestOptionalAttendees);
+    int requestDuration = (int) request.getDuration();
+    
+    Collection<TimeRange> allEventsTimeRanges = new ArrayList<TimeRange> ();
+    for (Iterator currEvent = events.iterator(); currEvent.hasNext(); ){ 
+      allEventsTimeRanges.add((TimeRange) ((Event) currEvent.next()).getWhen());
+    }
+
+    int startTime = 0;
+    int endTime = 0;
+    TimeRange tempTimeRange = TimeRange.fromStartEnd(startTime, endTime, true);
```

Comment:
nit: when calling an api or method that has a boolean (or other non-obvious parameter) you want to provide additional context so people know what the "true" here means. Two ways of doing that:
1) put true in a boolean variable named to describe the purpose of the parameter.
2) use small inline comment to describe the parameter's meaning (i.e. `startTime, endTime, /* isEndExclusive= */ true)`)